pkg/querier/querier.go

```diff
@@ -128,17 +129,27 @@ func getChunksIteratorFunction(cfg Config) chunkIteratorFunc {
 	return mergeChunks
 }
 
-func queryIngesters(queryIngestersWithin time.Duration, now time.Time, queryMaxT int64) bool {
+// QueryIngesters provides a check for whether the ingesters will be used for a given query.
+//
+// QueryIngesters is exposed for use before the constructing the distributor querier for a query,
+// in order to decide how to route queries or whether to construct a distributor querier at all;
+// queryIngestersClampMinT should be used for in-query logic.
+func QueryIngesters(queryIngestersWithin time.Duration, now time.Time, queryMaxT int64) bool {
```

Comment:
[nit] Maybe `ShouldQueryIngesters` and `ShouldQueryBlocksStore` would be clearer? (`QueryIngesters` sounds like it issues a query request to ingesters.)